File: src/main/java/named/entity/proprietoryModels/LingPipeExactDictionaryChunker.java
package named.entity.proprietoryModels;

import com.aliasi.dict.Dictionary;
import com.aliasi.dict.DictionaryEntry;
import com.aliasi.dict.ExactDictionaryChunker;
import com.aliasi.dict.MapDictionary;
import com.aliasi.tokenizer.IndoEuropeanTokenizerFactory;
import com.opencsv.CSVReader;
import preprocessing.Preprocessor;
import preprocessing.PreprocessorImp;

import java.io.FileReader;

public class LingPipeExactDictionaryChunker {

    public static ExactDictionaryChunker initializeDictionary(String fileName, String nerCategory) {
        final Dictionary<String> dictionary = getProfileDictionary(fileName, nerCategory);
        return new ExactDictionaryChunker(dictionary, IndoEuropeanTokenizerFactory.INSTANCE, true, false);
    }


    private static Dictionary<String> getProfileDictionary(String fileName, String nerCategory) {
        final MapDictionary<String> dictionary = new MapDictionary<>();
        try {
            FileReader filereader = new FileReader(LingPipeExactDictionaryChunker.class.getResource(fileName).getFile());
            CSVReader csvReader = new CSVReader(filereader);
            String[] nextRecord;
            Preprocessor preprocessor = new PreprocessorImp();
            while ((nextRecord = csvReader.readNext()) != null) {
                for (String cell : nextRecord) {
                    cell = preprocessor.getPreprocessedText(cell);
                    //profiles should not look like urls
                    cell = cell.replace("www.", "");
                    cell = cell.replace(".de", "");
                    cell = cell.replace(".com", "");
                    dictionary.addEntry(new DictionaryEntry<>(cell, nerCategory, 1.0));
                }
            }
        } catch (Exception e) {
            throw new IllegalStateException(e);
        }
        return dictionary;
    }
}

File: src/main/java/named/entity/proprietoryModels/LingPipeExactDictionaryChunker.kt
package named.entity.proprietoryModels

import com.aliasi.dict.Dictionary
import com.aliasi.dict.DictionaryEntry
import com.aliasi.dict.ExactDictionaryChunker
import com.aliasi.dict.MapDictionary
import com.aliasi.tokenizer.IndoEuropeanTokenizerFactory
import com.opencsv.CSVReader
import preprocessing.PreprocessorImp
import java.io.FileReader

object LingPipeExactDictionaryChunker {

    fun initializeDictionary(fileName: String, nerCategory: String): ExactDictionaryChunker {
        val dictionary = getProfileDictionary(fileName, nerCategory)
        return ExactDictionaryChunker(dictionary, IndoEuropeanTokenizerFactory.INSTANCE, true, false)
    }


    private fun getProfileDictionary(fileName: String, nerCategory: String): Dictionary<String> {
        val dictionary = MapDictionary<String>()
        try {
            val filereader = FileReader(LingPipeExactDictionaryChunker::class.java.getResource(fileName).file)
            val csvReader = CSVReader(filereader)
            val preprocessor = PreprocessorImp()
            var nextRecord: Array<String>
            while (csvReader.readNext() != null) {
                nextRecord = csvReader.readNext()
                for (cell in nextRecord) {
                    val processedCell =
                        preprocessor.getPreprocessedText(cell)
                            .replace("www.", "")
                            .replace(".de", "")
                            .replace(".com", "")
                    dictionary.addEntry(DictionaryEntry(processedCell, nerCategory, 1.0))
                }
            }
        } catch (e: Exception) {
            throw IllegalStateException(e)
        }

        return dictionary
    }
}
